mypy/semanal.py

```diff
@@ -587,6 +587,37 @@ def _visit_overloaded_func_def(self, defn: OverloadedFuncDef) -> None:
             # redefinitions already.
             return
 
+        # We know this is an overload def -- let's handle classmethod and staticmethod
+        class_status = []
+        static_status = []
+        for item in defn.items:
+            if isinstance(item, Decorator):
+                inner = item.func
+            elif isinstance(item, FuncDef):
+                inner = item
+            else:
+                raise AssertionError()
```

Comment:
Please add a message to all assertion errors. Typically we write `assert False, "Impossible blah-blah-blah"`.

Reply:
Done!